app/controllers/api/v1/debtors_controller.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module Api::V1
+  class DebtorsController < BaseOrganizationController
+    def index
+      debtors = []
+      total = 0
+      summ_by_currencies = []
+      current_organization.invoice_debtors.each do |debtor|
+        indebtedness = debtor.indebtedness
+        indebtedness.each do |debt|
+          amount = ConvertedMoneyPresenter.new(
+            debt,
+            current_organization.default_currency
+          ).present
+          debtors.push({
+            id: debtor.id,
+            name: debtor.name,
+            amount: amount
+          })
+          total += amount[:total]
+        end
+        indebtedness_summ!(summ_by_currencies, indebtedness)
+      end
+      render json: {
+        debtors: debtors,
+        summ_by_currencies: summ_by_currencies,
+        total: total
+      }.to_json
+    end
+
+    private
+
+    def indebtedness_summ!(summ, indebtedness)
+      indebtedness.each do |debt|
```

Comment:
@dissident and why can't you do it using sql `SUM` and `GROUP BY`? Check out `invoices_debt` method in `layout_helper.rb` which makes same calculations.